staging/src/k8s.io/apiserver/pkg/server/dynamiccertificates/client_ca.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dynamiccertificates
+
+import (
+	"bytes"
+)
+
+// CAContentProvider provides ca bundle byte content
+type CAContentProvider interface {
+	// Name is just an identifier
+	Name() string
```

Comment:
Do we only expect CAContentProvider implementations in this package? Maybe leave these methods private.

Reply:
> Do we only expect CAContentProvider implementations in this package? Maybe leave these methods private.

I'd like to have the option of writing an implementation downstream.  Imagine something similar to watching the kube-system namespace.